compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ExpressionCodegen.kt

```diff
@@ -825,9 +833,13 @@ class ExpressionCodegen(
         jump.markLineNumber(startOffset = true)
         val endLabel = Label()
         val stackElement = unwindBlockStack(endLabel, data) { it is LoopInfo && it.loop == jump.loop } as LoopInfo?
-            ?: throw AssertionError("Target label for break/continue not found")
-        mv.fixStackAndJump(if (jump is IrBreak) stackElement.breakLabel else stackElement.continueLabel)
-        mv.mark(endLabel)
+        if (stackElement == null) {
+            generateGlobalReturnFlagIfPossible(jump.loop.nonLocalReturnLabel(jump is IrBreak))
+            mv.areturn(Type.VOID_TYPE)
+        } else {
+            mv.fixStackAndJump(if (jump is IrBreak) stackElement.breakLabel else stackElement.continueLabel)
```

Comment:
Looks like `fixStackAndJump` should be also called in `then` branch in general case:  e.g. foo(param1, param2 ?: return@xxx), where return@xxx transformed to non-local break/continue

Reply:
This is true, but fixStackAndJump can only directly precede a jump instruction, so we can't generate it here yet. Instead, see the changes I made in MethodInliner: if the jump target is not the end of the inlined method, it will insert a fixStackAndJump and put any finally blocks before it.